lib/fennec/udp/worker.ex

```diff
@@ -66,22 +71,40 @@ defmodule Fennec.UDP.Worker do
 
   def handle_info({:udp, socket, ip, port, data}, state = %{turn:
                   %TURN{allocation: %TURN.Allocation{socket: socket}}}) do
-    _ = handle_peer_data(ip, port, data, state)
-    {:noreply, state}
+    now = Fennec.Helper.now
+    next_state =
+      case Map.get(state.turn.permissions, ip) do
+        nil ->
+          Logger.debug(~s"Dropped data from peer #{ip}:#{port} due to no permission")
+          __MODULE__.handle_peer_data(:no_permission, ip, port, data, state)
```

Comment:
Solely because `handle_peer_data` is mocked in tests and if it would be called locally, it wouldn't be called through mock.